secrets/store.go

```diff
@@ -59,62 +58,63 @@ func (s *Store) parser(r io.Reader) (interface{}, error) {
 		return nil, err
 	}
 
-	if s.secretHandlerFunc != nil {
-		s.secretHandlerFunc(secrets)
-	}
+	s.secretHandlerFunc(secrets)
 
 	return secrets, nil
 }
 
-// SecretHandler creates the middleware chain.
+// secretHandler creates the middleware chain.
 func (s *Store) secretHandler(middlewares ...SecretMiddleware) {
-	if s.secretHandlerFunc == nil {
-		s.secretHandlerFunc = nopSecretHandlerFunc
-	}
-
 	for _, m := range middlewares {
 		s.secretHandlerFunc = m(s.secretHandlerFunc)
 	}
 }
 
-// GetSimpleSecret loads secrets from watcher, and fetches a simple secret from secrets
-func (s *Store) GetSimpleSecret(path string) (SimpleSecret, error) {
-	secrets, ok := s.watcher.Get().(*Secrets)
-	if !ok {
-		return SimpleSecret{}, fmt.Errorf("unexpected type %T", secrets)
-	}
+func (s *Store) getSecrets() *Secrets {
+	return s.watcher.Get().(*Secrets)
+}
 
-	return secrets.GetSimpleSecret(path)
+// Stop stops the underlying filewatcher and release associated resources.
+//
+// After Stop is called, you won't get any updates to the secret file,
+// but can still access the secrets as they were before Stop is called.
+//
+// It's OK to call Stop multiple times. Calls after the first one are no-ops.
+func (s *Store) Stop() {
```

Comment:
I wonder if we can find a method name that fits the semantic more. I was a bit confused about stopping the store. It doesn't really stop the store but just the updating of the underlying file I guess?

Reply:
`StopUpdating`? `Release`? Do you have any suggestions?

Reply:
`Release` sounds good or maybe `Close`?

Reply:
In which scenario would this usually be called?

Reply:
defer this right after a store is initialized to have an clean exit.

Reply:
I mean for instance in production code. I saw the usage in test, I was just wondering it's purpose on the public interface of store :)

Reply:
You can still defer this in the main function in prod code.

Reply:
Now where I understand this better I think both `Close()` or `Stop()` are fine.

Reply:
Renamed to `Close`.